TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java

```diff
@@ -220,12 +222,21 @@ public DiagnosticTrackableCompletableFuture<String,Void> consumeBytes(ByteBuf pa
                     } else {
                         log.atInfo().setMessage(()->"Signaling previously returned CompletableFuture packet write had an exception : "
                                 + packetData + " hash=" + System.identityHashCode(packetData)).log();
+                        metricsLogger.atError(cause)
+                                .addKeyValue("channelId", channel.id().asLongText())
+                                .addKeyValue("requestId", diagnosticLabel)
```

Comment:
Have this key off of a UniqueRequestKey value.  The diagnosticLabel could be set to anything, including null and all of the rest of the code should work just fine.  If you want to log requestIds, take that in instead of a diagnosticId.  If you want to log diagosticIds, then relabel.
Of course - I don't like either of those options as much as injecting a dependency.  The same way that we take a diagnosticLabel which is a string that's simply prepended to our log statements, you can do something similar here (& elsewhere, as I imagine that this pattern will be quite common).  You can supply a decorator Function<LoggingEventBuilder,LoggingEventBuilder> to the c'tor that adds any additional key-value-bindings that the calling context would like to use.  This is pretty similar to the MDC approach earlier, but more tightly controlled.  You could even have the Function subsumed by a metricsLogger field (instead of a static) so that none of your calls within a class would need to change/do anything.